// accordian/dist/esm/accordian-component.entry.js
import { r as registerInstance, c as createEvent, h } from './index-8e328632.js';

const accordianComponentCss = "*{font-family:'Lato', sans-serif}.container{display:flex;flex-direction:column;justify-content:center;align-items:center}.accordion{cursor:pointer;padding:18px;text-align:left;border-radius:20px;font-size:1.2rem;font-weight:bold;outline:0}.open{display:block;height:auto;border-radius:20px;border:0.5px solid rgb(199, 197, 197);width:200px}p{padding:18px}.close{display:none}span{float:right}";

const AccordianComponent = class {
  constructor(hostRef) {
    registerInstance(this, hostRef);
    this.onToggle = createEvent(this, "onToggle", 7);
    this.toggle = false;
  }
  toggleComponent() {
    this.toggle = !this.toggle;
    this.onToggle.emit({ visible: this.toggle });
  }
  render() {
    return (h("div", null, h("button", { class: "accordion", style: {
        width: this.width,
        backgroundColor: this.color,
      }, onClick: () => this.toggleComponent() }, this.label, this.toggle ? h("span", null, "\u25B2") : h("span", null, "\u25BC")), h("div", { class: `content-box ${this.toggle ? 'open' : 'close'}`, style: { width: this.width } }, h("p", null, this.description))));
  }
};
AccordianComponent.style = accordianComponentCss;

export { AccordianComponent as accordian_component };
