// accordian/dist/components/accordian-component.js
import { proxyCustomElement, HTMLElement, createEvent, h } from '@stencil/core/internal/client';

const accordianComponentCss = "*{font-family:'Lato', sans-serif}.container{display:flex;flex-direction:column;justify-content:center;align-items:center}.accordion{cursor:pointer;padding:18px;text-align:left;border-radius:20px;font-size:1.2rem;font-weight:bold;outline:0}.open{display:block;height:auto;border-radius:20px;border:0.5px solid rgb(199, 197, 197);width:200px}p{padding:18px}.close{display:none}span{float:right}";

const AccordianComponent$1 = /*@__PURE__*/ proxyCustomElement(class extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.__attachShadow();
    this.onToggle = createEvent(this, "onToggle", 7);
    this.toggle = false;
  }
  toggleComponent() {
    this.toggle = !this.toggle;
    this.onToggle.emit({ visible: this.toggle });
  }
  render() {
    return (h("div", null, h("button", { class: "accordion", style: {
        width: this.width,
        backgroundColor: this.color,
      }, onClick: () => this.toggleComponent() }, this.label, this.toggle ? h("span", null, "\u25B2") : h("span", null, "\u25BC")), h("div", { class: `content-box ${this.toggle ? 'open' : 'close'}`, style: { width: this.width } }, h("p", null, this.description))));
  }
  static get style() { return accordianComponentCss; }
}, [1, "accordian-component", {
    "label": [1],
    "description": [1],
    "width": [1],
    "color": [1],
    "toggle": [32]
  }]);
function defineCustomElement$1() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["accordian-component"];
  components.forEach(tagName => { switch (tagName) {
    case "accordian-component":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, AccordianComponent$1);
      }
      break;
  } });
}

const AccordianComponent = AccordianComponent$1;
const defineCustomElement = defineCustomElement$1;

export { AccordianComponent, defineCustomElement };
